File: app/src/main/java/com/BlizzardArmory/overwatch/QuickPlay/Combat.java

package com.BlizzardArmory.overwatch.QuickPlay;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Combat {

    @SerializedName("barrierDamageDone")
    @Expose
    private int barrierDamageDone;
    @SerializedName("damageDone")
    @Expose
    private int damageDone;
    @SerializedName("deaths")
    @Expose
    private int deaths;
    @SerializedName("eliminations")
    @Expose
    private int eliminations;
    @SerializedName("environmentalKills")
    @Expose
    private int environmentalKills;
    @SerializedName("finalBlows")
    @Expose
    private int finalBlows;
    @SerializedName("heroDamageDone")
    @Expose
    private int heroDamageDone;
    @SerializedName("meleeFinalBlows")
    @Expose
    private int meleeFinalBlows;
    @SerializedName("multikills")
    @Expose
    private int multikills;
    @SerializedName("objectiveKills")
    @Expose
    private int objectiveKills;
    @SerializedName("objectiveTime")
    @Expose
    private String objectiveTime;
    @SerializedName("soloKills")
    @Expose
    private int soloKills;
    @SerializedName("timeSpentOnFire")
    @Expose
    private String timeSpentOnFire;

    public int getBarrierDamageDone() {
        return barrierDamageDone;
    }

    public void setBarrierDamageDone(int barrierDamageDone) {
        this.barrierDamageDone = barrierDamageDone;
    }

    public int getDamageDone() {
        return damageDone;
    }

    public void setDamageDone(int damageDone) {
        this.damageDone = damageDone;
    }

    public int getDeaths() {
        return deaths;
    }

    public void setDeaths(int deaths) {
        this.deaths = deaths;
    }

    public int getEliminations() {
        return eliminations;
    }

    public void setEliminations(int eliminations) {
        this.eliminations = eliminations;
    }

    public int getEnvironmentalKills() {
        return environmentalKills;
    }

    public void setEnvironmentalKills(int environmentalKills) {
        this.environmentalKills = environmentalKills;
    }

    public int getFinalBlows() {
        return finalBlows;
    }

    public void setFinalBlows(int finalBlows) {
        this.finalBlows = finalBlows;
    }

    public int getHeroDamageDone() {
        return heroDamageDone;
    }

    public void setHeroDamageDone(int heroDamageDone) {
        this.heroDamageDone = heroDamageDone;
    }

    public int getMeleeFinalBlows() {
        return meleeFinalBlows;
    }

    public void setMeleeFinalBlows(int meleeFinalBlows) {
        this.meleeFinalBlows = meleeFinalBlows;
    }

    public int getMultikills() {
        return multikills;
    }

    public void setMultikills(int multikills) {
        this.multikills = multikills;
    }

    public int getObjectiveKills() {
        return objectiveKills;
    }

    public void setObjectiveKills(int objectiveKills) {
        this.objectiveKills = objectiveKills;
    }

    public String getObjectiveTime() {
        return objectiveTime;
    }

    public void setObjectiveTime(String objectiveTime) {
        this.objectiveTime = objectiveTime;
    }

    public int getSoloKills() {
        return soloKills;
    }

    public void setSoloKills(int soloKills) {
        this.soloKills = soloKills;
    }

    public String getTimeSpentOnFire() {
        return timeSpentOnFire;
    }

    public void setTimeSpentOnFire(String timeSpentOnFire) {
        this.timeSpentOnFire = timeSpentOnFire;
    }

}

File: app/src/main/java/com/BlizzardArmory/model/overwatch/account/Combat.kt
package com.BlizzardArmory.model.overwatch.account

import com.google.gson.annotations.SerializedName


import java.text.DecimalFormat
import java.text.NumberFormat
import java.util.*

/**
 * The type Combat.
 */

data class Combat(

    @SerializedName("barrierDamageDone")
    var barrierDamageDone: Double,

    @SerializedName("damageDone")
    var damageDone: Double,

    @SerializedName("deaths")
    var deaths: Double,

    @SerializedName("eliminations")
    var eliminations: Double,

    @SerializedName("environmentalKills")
    var environmentalKills: Double,

    @SerializedName("finalBlows")
    var finalBlows: Double,

    @SerializedName("heroDamageDone")
    var heroDamageDone: Double,

    @SerializedName("meleeFinalBlows")
    var meleeFinalBlows: Double,

    @SerializedName("multikills")
    var multikills: Double,

    @SerializedName("objectiveKills")
    var objectiveKills: Double,

    @SerializedName("objectiveTime")
    var objectiveTime: String,

    @SerializedName("soloKills")
    var soloKills: Double,

    @SerializedName("timeSpentOnFire")
    var timeSpentOnFire: String
) {
    val combat: HashMap<String, String>
        get() {
            val combatList = HashMap<String, String>()
            val formatter: NumberFormat = DecimalFormat("#0")
            combatList["BARRIER DAMAGE DONE"] = formatter.format(barrierDamageDone)
            combatList["DAMAGE DONE"] = formatter.format(damageDone)
            combatList["DEATHS"] = formatter.format(deaths)
            combatList["ELIMINATIONS"] = formatter.format(eliminations)
            combatList["ENVIRONMENTAL KILLS"] = formatter.format(environmentalKills)
            combatList["FINAL BLOWS"] = formatter.format(finalBlows)
            combatList["HERO DAMAGE DONE"] = formatter.format(heroDamageDone)
            combatList["MELEE FINAL BLOWS"] = formatter.format(meleeFinalBlows)
            combatList["MULTIKILLS"] = formatter.format(multikills)
            combatList["OBJECTIVE KILLS"] = formatter.format(objectiveKills)
            combatList["OBJECTIVE TIME"] = objectiveTime
            combatList["SOLO KILLS"] = formatter.format(soloKills)
            combatList["TIME SPENT ON FIRE"] = timeSpentOnFire
            return combatList
        }
}